api/resources/Resource.py

```diff
@@ -59,7 +59,7 @@ def get_response(self):
         if not response:
             if 'POST' in self.method_type:
                 response = self.request.app().resolve(getattr(self, 'create'))
-            elif 'GET' in self.method_type and '@' in self.route_url:
+            elif 'GET' in self.method_type and '@id' in self.route_url:
```

Comment:
i'm not entirely sold on the idea of hard coding `@id` into here .. what id you want `@slug`?

Reply:
The thing is, when your route is `/api/articles/@slug/comments` that resolve to `CommentResource` when you hit get request for `/api/articles/@slug/comments/` it gives the error about id cause as per code when you first checks @ in url defined so it goes show method instead of index.

Reply:
right .. i wonder if there is a diff check then we should be looking for 🤔

Reply:
We can check in reverse order if `@id` exists or `@` or regex check for string which ends with `@*`?

Reply:
yeah .. that might work. The issue is that `/api/articles/@slug/comments/` is actually getting the relationship though right? I don't think i ever supported that

Reply:
i think we need to rethink a simple if statement

Reply:
Relationship as in ? Here is my routes and it works as expected